google/cloud/sqlalchemy_spanner/sqlalchemy_spanner.py

```diff
@@ -48,7 +48,8 @@
 @listens_for(Pool, "reset")
 def reset_connection(dbapi_conn, connection_record):
     """An event of returning a connection back to a pool."""
-    dbapi_conn.connection.staleness = None
+    if getattr(dbapi_conn.connection, "staleness") is not None:
```

Comment:
Would we need any additional tests for this? I see a few attempts in https://github.com/googleapis/python-spanner-sqlalchemy/pull/187/commits/dfdbc7c21514372bb511fc1e81a2b2cd28a3394b etc, but that PR is closed.

Reply:
Attemps? I've tried to split one test class into two to check if there is an emulator-related issue there.
Additional check is added into the existing case with this PR.

Reply:
Ah, got it. Thanks for explaining what you were trying in that PR/commit.

Reply:
lgtm